session/create_dto.go

```diff
@@ -32,19 +33,41 @@ var (
 
 // CreateRequest structure represents message from service consumer to initiate session for given proposal id
 type CreateRequest struct {
-	ProposalId int             `json:"proposal_id"`
-	Config     json.RawMessage `json:"config"`
+	ProposalID   int             `json:"proposal_id"`
+	Config       json.RawMessage `json:"config"`
+	ConsumerInfo *ConsumerInfo   `json:"consumer_info,omitempty"`
+}
+
+// PaymentInfo represents the payment information that the provider has about the consumer
+type PaymentInfo struct {
+	LastPromise LastPromise `json:"lastPromise"`
+	FreeCredit  uint64      `json:"freeCredit"`
+}
+
+// LastPromise represents the last known promise to the provider
+// If the seqid and amount are 0 - there's no known info
+type LastPromise struct {
+	SequenceID uint64 `json:"sequenceID"`
+	Amount     uint64 `json:"amount"`
 }
 
 // CreateResponse structure represents service provider response to given session request from consumer
 type CreateResponse struct {
 	Success bool       `json:"success"`
 	Message string     `json:"message"`
 	Session SessionDto `json:"session"`
+	// Keeping this as a pointer for maximum backwards compatibility
+	PaymentInfo *PaymentInfo `json:"paymentInfo,omitempty"`
 }
 
 // SessionDto structure represents session information data within session creation response (session id and configuration options for underlying service type)
 type SessionDto struct {
 	ID     ID              `json:"id"`
 	Config json.RawMessage `json:"config"`
 }
+
+// ConsumerInfo represents the consumer related information
+type ConsumerInfo struct {
+	MystClientVersion string            `json:"mystClientVersion"`
```

Comment:
Cannot find where this 'MystClientVersion' is used. Need clarification and probably a separate task if its gonna be used. Payments itself should not depend on consumer version. If we would support more then one payment method (or protocol) they I would say we would need this version.

Reply:
You're correct - it's not used anymore. Was introduced here as we'll probably want to know what consumer version is on, and maybe adjust provider behaviour accordingly. Since we were already touching this session creation it only seemed appropriate to add this field.